infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/WaitSnapshotApplyState.java

```diff
@@ -133,7 +147,13 @@ public LogReplicationState processEvent(LogReplicationEvent event) throws Illega
                 log.debug("Shutdown Log Replication while waiting for snapshot sync apply to complete id={}", transitionEventId);
                 return fsm.getStates().get(LogReplicationStateType.ERROR);
             default: {
-                log.warn("Unexpected log replication event {} when in wait snapshot sync apply state.", event.getType());
+                if (!fsm.isValidTransition(transitionEventId, event.getMetadata().getRequestId())) {
+                    log.warn("Ignoring log replication event {} for sync {} when in wait snapshot sync apply state for sync {}",
+                            event.getType(), event.getMetadata().getRequestId(), transitionEventId);
+                    return this;
+                }
+                log.warn("Unexpected log replication event {} for sync {} when in wait snapshot sync apply state for sync {}",
+                        event.getType(), event.getMetadata().getRequestId(), transitionEventId);
```

Comment:
I think this is not correct, from the code itself we do something like: if it's not a valid transition, we hold on for current state; while if it's indeed valid, we throw out an exception...

also, it doesn't even fix the issue we noticed in the original scenario, if let's say it's a `SNAPSHOT_SYNC_CONTINUE` event from another snapshot sync, fsm is still stuck in `WaitSnapshotApplyState`

Reply:
The way I understand it is - 'if the transition was not meant for this fsm, ignore it and continue in the current state.'  Isnt it correct?

Reply:
Yes that is correct, @xcchang and I looked over this part yesterday just to confirm.